src/test/java/springboot/webapp/usersmanager/services_tests/UserServiceTests.java

```diff
@@ -0,0 +1,144 @@
+package springboot.webapp.usersmanager.services_tests;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.*;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import springboot.webapp.usersmanager.controllers.UserController;
+import springboot.webapp.usersmanager.entities.Role;
+import springboot.webapp.usersmanager.entities.User;
+import springboot.webapp.usersmanager.repositories.UserRepository;
+import springboot.webapp.usersmanager.services.UserService;
+
+import java.util.*;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(UserController.class)
+public class UserServiceTests {
+    @Autowired
```

Comment:
try to avoid Autowiring

Reply:
in this context i don't see any other solution